### PR TITLE
Keep map selection when searching

### LINKED ISSUE #3092: Typing in map search clears map selection and name preview
**Describe the bug**
Typing in the search box clears the map name in the preview on the right, and the map selection in the list.
Search Bug

**Expected behavior**
The map name and map selection should remain.

**OS**
Windows

### PR DESCRIPTION
**Pull request by obydog002**
Fix #3092 

Now retains the selected option

### CODE REVIEW

**Review by Sheikah45 — CHANGES_REQUESTED** [review]
Instead of introducing a mutable variable that we have to manually remember to set and track, I think it would be better to use the biconsumer version of subscribe on the selectedItem property in initMapSelection. This provides the new and old selected values, then if the new value is null we can select the old value if it is contained by filtered maps.

Effectively moving that logic outside of the setSelectedMap function.

Which is better for the long term as well since most of those properties set in setSelected map will be changed to be bound to derived properties eventually.

**Review by Sheikah45 — DISMISSED** [review]
The old item null check is superfluous but that is fine

File: src/main/java/com/faforever/client/game/CreateGameController.java [hunk 1]
```diff
@@ -108,6 +108,8 @@ public class CreateGameController extends NodeController<Pane> {
 
   private final SimpleInvalidationListener createButtonStateListener = this::setCreateGameButtonState;
 
+  private MapVersionBean lastMapChosen = null;
+
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/game/CreateGameController.java]: This field can be removed, also did you want to write a test for this?

Comment (obydog002) [reply src/main/java/com/faforever/client/game/CreateGameController.java]: Oh yep Ill remove that and add some tests

**Review by Sheikah45 — APPROVED** [review]

**Pull request merged**